src/main/java/org/jenkinsci/plugins/tokenmacro/MacroEvaluationException.java

```diff
@@ -30,7 +32,26 @@
  * @author Kohsuke Kawaguchi
  */
 public class MacroEvaluationException extends Exception {
+    
+    private final @CheckForNull String macroName;
+    
     public MacroEvaluationException(String message) {
+        this(message, null);
+    }
+
+    public MacroEvaluationException(String macroName, @CheckForNull String message) {
         super(message);
+        this.macroName = macroName;
+    }
+
+    @CheckForNull
+    public String getMacroName() {
+        return macroName;
+    }
+
+    @Override
+    public String getMessage() {
+        final String prefix = macroName != null ? "In "+macroName + ": " :  "";
+        return  prefix +  super.getMessage();
```

Comment:
This breaks one of the tests that checks for a message, see the build on buildhive for info.